docs/docs/sample-scripts/spo/report-on-sharepoint-files-containing-client-id-secret.md

```diff
@@ -0,0 +1,112 @@
+# Using CLI for Microsoft 365 and PowerShell to report on SharePoint Document Library files containing Client Id or Secret
+
+Author: [Joseph Velliah](https://sprider.blog/report-on-sharepoint-files-containing-pii)
+
+The big challenge with using APIs that require authentication in your JavaScript is that you’re forced to expose your API credentials to use them. Anyone who knows how to view source or view requests in their browser’s Developer Tools can view those credentials, steal them, and use them to access the API as you.
+
+This script can be used to scan and report on SharePoint Document Library files (specified file types) containing Client Id or Secret. Feel free to add additional patterns to match your requirements.
+
+Prerequisites:
+
+- [CLI for Microsoft 365](https://pnp.github.io/cli-microsoft365/)
+- SharePoint Online Document Library files(aspx, js, html) containing dummy/real Client Id or Secrets
+
+```powershell tab="PowerShell Core"
+# This script is created to support only the following file extensions.
+$supportedFileExtensions = "aspx", "js", "html"
+$spolHostName = "https://tenant-name.sharepoint.com"
+$spolSiteRelativeUrl = "/sites/site-name"
+$spolDocLibTitle = "library-title"
+$resultDir = "Output"
+
+
+$Patterns = @()
+
+$Pattern0 = @{
+    Title       = "Client ID"
+    Regex       = "(\{){0,1}[0-9a-fA-F]{8}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{4}\-[0-9a-fA-F]{12}(\}){0,1}"
```

Comment:
Is there any other way to detect client ID? I'm afraid reporting every GUID will yield a lot of false positive making the report of little use. Have you had a look at the Microsoft Application Inspector (https://github.com/microsoft/ApplicationInspector) and what it's detecting?

Reply:
> Is there any other way to detect client ID? I'm afraid reporting every GUID will yield a lot of false positive making the report of little use. Have you had a look at the Microsoft Application Inspector (https://github.com/microsoft/ApplicationInspector) and what it's detecting?

@waldekmastykarz - Application Inspector is a dotnet tool package. Could you please clarify on the integration part of this?

Reply:
> > Is there any other way to detect client ID? I'm afraid reporting every GUID will yield a lot of false positive making the report of little use. Have you had a look at the Microsoft Application Inspector (https://github.com/microsoft/ApplicationInspector) and what it's detecting?
> 
> @waldekmastykarz - Application Inspector is a dotnet tool package. Could you please clarify on the integration part of this? Ran this tool against a js file with dummy client id and secrets in it but i am not getting any good insights. 

Please check the output below: 

![Screen Shot 2020-09-22 at 10 40 46 AM](https://user-images.githubusercontent.com/21008316/93905179-47fe7400-fcc0-11ea-9cf4-8e11c7dca588.png)

Reply:
I didn't quite mean integrating with it. Instead, I'd suggested to look at its rules and the patterns it uses. Majority of the rules are regex based and the patterns could be applied in this sample without pulling in the whole tool.